app/models/graph/types/query.rb

```diff
@@ -46,6 +46,17 @@ module Types
         }
       end
 
+      field :viewer, Graph::Types::User, 'The currently authenticated user (if any)' do
+        resolve ->(_, _, ctx) {
+          case ctx[:user]
+          when ::User
```

Comment:
Maybe I should change this code a bit.

When someone is not logged in, I store `GuestUser.new` in `ctx[:user]`, but in this specific case I think we want to resolve to `nil` when the person is not logged in.

Maybe a better approach would be to define `def logged_in?` in `GuestUser` and `User` such that we can simply do `ctx[:user] if ctx[:user].logged_in?`.

We can use the same logic in our mutations.